libimage/runtime.go

```diff
@@ -541,6 +541,14 @@ type RemoveImagesOptions struct {
 	// using a removed image.  Use RemoveContainerFunc for a custom logic.
 	// If set, all child images will be removed as well.
 	Force bool
+	// If set, do not look for items/instances in the manifest list that
+	// match the current platform but return the manifest list as is.
+	// only check for manifest list, return ErrNotAManifestList if not found.
+	LookupManifest bool
```

Comment:
I don't understand why we need these two options.  It's already confusing me (and I wrote most of libimage). Imagine how a new users would approach it.

Isn't `ManifestList=true` plus `NoSuchImage` effectively the same thing as `LookupManifest=true`?

Reply:
@vrothberg ah we can change names if its confusing but we discussed the use case here https://github.com/containers/common/pull/747#discussion_r697286818

Reply:
I tried explaining use case in this PR https://github.com/containers/common/pull/753#discussion_r699996500

Reply:
If we need two options, then it looks like we're fixing two different issues (which should be at least two different commits). From what I see, I think we only need one option to address `manifest rm`.

From #753 (comment)
> However buildah rmi <name> can remove manifest list if found otherwise it should search for images so we call ImageLookup with ManifestList: true

That is a separate issue and hasn't been discussed yet.  I'd also strongly oppose it for now, since `rmi` is supposed to remove images not manifests.

So for now, let's limit this PR to `manifest rm` only.

Did you consider adding a `(*ManifestList).Remove()` method?  I think this would less confusing.  So far, I tried to keep ManifestLists separate entities in libimage as I don't want to mix it too much.

Reply:
@vrothberg I agree discussed same here https://github.com/containers/buildah/issues/3490#issuecomment-910033675